### ISSUE TITLE
`PyTorchModelHubMixin` models with safetensors weights are loaded in training mode

### ISSUE THREAD

**Issue by mxmpl** [huggingface/huggingface_hub#3981]
### Describe the bug

`PyTorchModelHubMixin` documentation says that "The model is set in evaluation mode by default using `model.eval()`" but this is only true [when it is loaded with `_load_as_pickle`](https://github.com/huggingface/huggingface_hub/blob/d40f2ed861821c3702a7768e2df3d07703142913/src/huggingface_hub/hub_mixin.py#L806).

When `_load_as_safetensor` is used, which is the default if the user has previously called `save_pretrained`, the model is loaded in **training mode**.
This has quite bad side-effects.

The fix is simply to add `model.eval()` before the return in `_load_as_safetensor`.

### Reproduction

```py
import torch
from huggingface_hub import PyTorchModelHubMixin
from torch import nn


class Model(nn.Module, PyTorchModelHubMixin):
    def __init__(self):
        super().__init__()
        self.d = nn.Dropout(1)

    def forward(self, x):
        return self.d(x)


m1 = Model()
m1.save_pretrained("out")

m2 = Model.from_pretrained("out")
print(m2.training)  # True

x = torch.ones((1, 10))
print(m2(x))  # tensor([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]])

m2.eval()
print(m2(x))  # tensor([[1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]])
```

### Logs

```shell

```

### System info

```shell
- huggingface_hub version: 1.8.0
- Platform: macOS-15.7.4-arm64-arm-64bit-Mach-O
- Python version: 3.13.12
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Running in Google Colab Enterprise ?: No
- Token path ?: /Users/mpoli/.cache/huggingface/token
- Has saved token ?: True
- Who am I ?: mpoli
- Configured git credential helpers: osxkeychain, store
- Installation method: unknown
- httpx: 0.28.1
- hf_xet: 1.4.2
- gradio: N/A
- tensorboard: N/A
- ENDPOINT: https://huggingface.co
- HF_HUB_CACHE: /Users/mpoli/.cache/huggingface/hub
- HF_ASSETS_CACHE: /Users/mpoli/.cache/huggingface/assets
- HF_TOKEN_PATH: /Users/mpoli/.cache/huggingface/token
- HF_STORED_TOKENS_PATH: /Users/mpoli/.cache/huggingface/stored_tokens
- HF_HUB_OFFLINE: False
- HF_HUB_DISABLE_TELEMETRY: False
- HF_HUB_DISABLE_PROGRESS_BARS: None
- HF_HUB_DISABLE_SYMLINKS_WARNING: False
- HF_HUB_DISABLE_EXPERIMENTAL_WARNING: False
- HF_HUB_DISABLE_IMPLICIT_TOKEN: False
- HF_HUB_DISABLE_XET: False
- HF_HUB_ETAG_TIMEOUT: 10
- HF_HUB_DOWNLOAD_TIMEOUT: 10
- HF_XET_HIGH_PERFORMANCE: False
```

**Comment by Wauplin** [comment]
Thanks for the report @mxmpl !

@cursor can you open a PR to fix this (no need to add a test)
